app/models/search_builder.rb

```diff
@@ -1,3 +1,12 @@
 class SearchBuilder < Blacklight::SearchBuilder
   include Blacklight::Solr::SearchBuilderBehavior
+
+  def add_work_id solr_params
+    if blacklight_params[:search_field] == 'leaf' && blacklight_params[:workid].present?
+      solr_params[:fq] ||= []
+      workid = blacklight_params[:workid]
+      workid = "#{workid}*" unless workid.include? '*'
+      solr_params[:fq] << "part_of_ssi:#{workid}"
+    end
+  end
 end
```

Comment:
I think you could put this method in the CatalogController as well, but perhaps it's better practice to put it here? In that case @xenma's  branch should be refactored a bit to put the search_params logic in the same place...